packages/keyring/src/pair/index.ts

```diff
@@ -44,7 +44,13 @@ const TYPE_SIGNATURE = {
 const TYPE_ADDRESS = {
   ecdsa: (p: Uint8Array) => p.length > 32 ? blake2AsU8a(p) : p,
   ed25519: (p: Uint8Array) => p,
-  ethereum: (p: Uint8Array) => keccakAsU8a(secp256k1Expand(p)),
+  ethereum: (p: Uint8Array) => {
+    if (p.length === 20) {
+      return (p);
+    } else {
+      return hexToU8a(ethereumEncode(keccakAsU8a(secp256k1Expand(p))));
```

Comment:
Since Ethereum addresses are with the checksum encoding, I would not change this. It adds value in the ecosystem.

Reply:
Not change what?

Reply:
The encoding. So we updated all address outputs here to be non-checksummed. In the Ethereum ecosystem the checksums do add value. If we want non checksum addresses (which we certainly don't), the correct place would be to encode it without. In this case however, we don't want to do this.

1. the address output should output the U8a - only a simple `p.length === 20 ? p : keccakAsU8a(secp256k1Expand(p))`
2. the encodeaddress should use ethereumEncode

So effectively we will probably only have the line change in 1 as code-changes (for length 20, return as-is, else keep the status quo) and then your new tests to ensure it is working

Reply:
Sounds good, I pushed a commit in that sens